addons/mail/models/ir_translation.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, _
+from odoo.exceptions import AccessError
+
+
+class IrTranslation(models.Model):
+    _inherit = 'ir.translation'
+
+    def create(self, vals_list):
+        translations = super().create(vals_list)
+        translations._check_is_dynamic()
+        return translations
+
+    def write(self, vals):
+        res = super().write(vals)
+        self._check_is_dynamic()
+        return res
+
+    def _check_is_dynamic(self):
+        # if we don't modify translation of at least a model that inherits from mail.render.mixin, we ignore it
+        # translation.name can be a path, and so not in the pool, so type(None) will exclude these translations.
+        translations_for_mail_render_mixin = self.filtered(
+            lambda translation: issubclass(type(self.env.get(translation.name.split(',')[0])), self.pool['mail.render.mixin'])
+        )
+        if not translations_for_mail_render_mixin:
+            return
+
+        # if we are admin, or that we can update mail.template we ignore
+        if self.env.is_admin() or self.env.user.has_group('mail.group_mail_template_editor'):
```

Comment:
can we move this at the beginning of the method ?
(so we skip everything if we are admin)

Reply:
I did it, but now we will do query has_group (probably already in cache) on each create/write before to check if we write on field that require this group :thinking: 

I don't think that the issubclass is heavy... 
Not sure that the move is smart. what do you think about ?

Reply:
hum indeed :thinking: but in the filtered you do translation.name, but it should be in the cache as well
both are ok for me

Reply:
translation is already in cache since it is called after the create and the write ;)
So yes, I will move it back after, since it is ok for you too.
It will avoid a random extra query on each write of ir.translation.